static/js/plugin_loading_and_search.js

```diff
@@ -0,0 +1,169 @@
+document.addEventListener("DOMContentLoaded", function () {
+  let currentSearchAbortController = null;
+
+  loadPlugins();
+
+  async function loadPlugins() {
+    const pluginContainer = document.getElementById("pluginContainer");
+    pluginContainer.insertBefore(
+      makeSpinner("Loading plugins..."),
+      pluginContainer.firstChild,
+    );
+    const response = await fetch("plugins_list.html");
+    const html = await response.text();
+    pluginContainer.innerHTML = html;
+    updatePluginCount();
+
+    // attach search listener after plugins are loaded
+    attachSearchListener();
+  }
+
+  function updatePluginCount() {
+    // Select all <a> tags within the pluginContainer
+    const allPlugins = document
+      .getElementById("pluginContainer")
+      .getElementsByTagName("a");
+
+    // Filter the list of all <a> tags to only those that are currently displayed
+    const visiblePlugins = Array.from(allPlugins).filter(
+      (plugin) => plugin.style.display !== "none",
+    );
+
+    // Update the displayed count with the number of visible plugins
+    document.getElementById("pluginCount").textContent = visiblePlugins.length;
+  }
+
+  let searchTimeout;
+
+  function attachSearchListener() {
+    const searchBox = document.getElementById("searchBox");
+    searchBox.addEventListener("input", function (event) {
+      clearTimeout(searchTimeout);
+      searchTimeout = setTimeout(() => {
+        handleSearch(event.target.value);
+      }, 300);
+    });
+
+    // trigger search if user already typed something
+    if (searchBox.value.trim()) {
```

Comment:
I think this searches with every character, right? Maybe we should not launch a search until like, three or four characters are typed?